src/url_search_params.cpp

```diff
@@ -0,0 +1,59 @@
+#include "ada.h"
+#include "ada/url_search_params.h"
+
+#include <optional>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace ada {
+
+void url_search_params::remove(const std::string_view key) {
+  params.erase(
+      std::remove_if(params.begin(), params.end(),
+                     [&key](auto &param) { return std::get<0>(param) == key; }),
+      params.end());
+}
+
+void url_search_params::remove(const std::string_view key,
+                               std::string_view value) {
+  params.erase(std::remove_if(params.begin(), params.end(),
+                              [&key, &value](auto &param) {
+                                return std::get<0>(param) == key &&
+                                       std::get<1>(param) == value;
+                              }),
+               params.end());
+}
+
+void url_search_params::sort() const noexcept {
+  // TODO: Implement this
```

Comment:
it is almost std::sort(params.begin(), params.end()), but not quite.

Reply:
The following might do...

```C++
std::stable_sort(params.begin(), params.end(), [](const std::tuple<std::string,std::string>& a, const std::tuple<std::string,std::string>& b) {
 return std::get<0>(a) > std::get<0>(b);
});
```

You need the `algorithm` header.

Reply:
Yes, with a slight bit of change, this worked:

```
void url_search_params::sort() {
  typedef std::pair<std::string, std::string> key_value_pair;
  std::stable_sort(params.begin(), params.end(),
                   [](const key_value_pair &lhs, const key_value_pair &rhs) {
                     return lhs.first > rhs.first;
                   });
}
```